pageserver/src/http/routes.rs

```diff
@@ -117,6 +117,13 @@ fn apierror_from_tenant_map_insert_error(e: TenantMapInsertError) -> ApiError {
     }
 }
 
+fn apierror_from_tenant_state_error(err: TenantStateError) -> ApiError {
```

Comment:
```suggestion
fn api_error_from_tenant_state_error(err: TenantStateError) -> ApiError {
```

Could that be an `impl From<TenantStateError> for ApiError`?
Plus same `.into()` -> `T::from` nit style comment.

Reply:
kept the same apierror as rest of the code

Reply:
I don't understand the response:
> kept the same apierror as rest of the code

Why was this done?

Reply:
the naming of all other apierror functions has apierror prefix and not api_error

Reply:
Aah yes, I was thinking more about the other suggestions (From impl, .into() => T::from) (on the same thread starting message).

Reply:
renamed all

Reply:
> Could that be an impl From<TenantStateError> for ApiError?

This remained unanswered. I think we can keep current way of things at least for now. It can be a separate PR to clean this up. I think @SomeoneToIgnore meant to have following impls instead of these functions

```rust
impl From<PageReconstructError> for ApiError {
    // ...
}

impl From<TenantMapInsertError> for ApiError {
    // ...
}

impl From<TenantStateError> for ApiError {
    // ...
}
```

This can be a small tech_debt issue that you can pick up at any time

Reply:
done :-)